sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -98,8 +98,12 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
       ctx.identifier != null &&
       ctx.identifier.getText.toLowerCase == "noscan") {
       AnalyzeTableCommand(visitTableIdentifier(ctx.tableIdentifier).toString)
-    } else {
+    } else if (ctx.identifierSeq() == null) {
```

Comment:
As mentioned in  [the comment](https://github.com/apache/spark/pull/15090#r78687294), we are going to change the "ANALYZE" syntax in SqlBase.g4, i.e. make the identifierSeq non-optional, which it's different from Hive. Is this ok? @rxin @hvanhovell